client/src/test/java/io/pravega/client/control/impl/ControllerImplTest.java

```diff
@@ -2541,5 +2542,39 @@ public void testDeleteKeyValueTable() {
         assertFalse(deleteKVTableStatus.join());
     }
 
+    @Test
+    public void testGetEndPointForSegment() throws Exception {
+        CompletableFuture<PravegaNodeUri> endpointForSegment;
+        // reads from network and store in cache
+        endpointForSegment = controllerClient.getEndpointForSegment("scope1/stream1/0");
+        assertEquals(new PravegaNodeUri("localhost", SERVICE_PORT), endpointForSegment.get());
+
+        // picks from cache
+        endpointForSegment = controllerClient.getEndpointForSegment("scope1/stream1/0");
+        assertEquals(new PravegaNodeUri("localhost", SERVICE_PORT), endpointForSegment.get());
+
+        // for the case when entry in cache is expired
+        Thread.sleep(21000);
```

Comment:
Sleeping is not allowed in unit tests. Please make the test not time dependent.

Reply:
Incorporated changes